biorxiv/polka_subset_experiment/04_regression_results_post_covid_preprints.py

```diff
@@ -0,0 +1,302 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:annorxiver]
+#     language: python
+#     name: conda-env-annorxiver-py
+# ---
+
+# # Re-Run Analyses with Polka et. al. Subset
+
+# This notebook was created in response to Polka et al. Group's inquiry on training a logistic regression model on preprints posted recently rather than preprints from 2019 and below.
+# Overall their subset can be separated with a few features.
+
+# +
+from pathlib import Path
+import sys
+
+from gensim.models import Word2Vec
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+import numpy as np
+import pandas as pd
+import plotnine as p9
+import requests
+from scipy.spatial.distance import cdist
+from scipy.stats import linregress
+from sklearn.model_selection import GridSearchCV
+from sklearn.linear_model import LogisticRegressionCV, LogisticRegression
+from sklearn.preprocessing import StandardScaler
+from sklearn.tree import DecisionTreeClassifier, export_graphviz
+import spacy
+import tqdm
+
+from annorxiver_modules.document_helper import generate_doc_vector
+
+mpl.rcParams["figure.dpi"] = 250
+# -
+
+# # Random BioRxiv Sample
+
+manual_papers_df = pd.read_csv(str(Path("output/all_pairs_2021-02-11.csv")))
+manual_papers_df.head().T
+
+api_url = "https://api.biorxiv.org/details/biorxiv/2020-01-01/2020-04-30"
+response = requests.get(api_url)
+content = response.json()
+total_papers = content["messages"][0]["total"]
+total_papers
+
+np.random.seed(100)
+selected_biorxiv_papers = np.random.randint(0, total_papers, 100)
+selected_biorxiv_papers.sort()
+selected_biorxiv_papers
+
+paper_cursor = {}
```

Comment:
>Sorry I"m confused at what this dictionary is representing. I"m assuming these are a random selection of 100 papers across some years where the keys are the years in integer form. Is that right?

Yes bioRxiv's API list papers posted within a timeframe as 100 papers per page. This dictionary is design to keep track of which page each select paper was posted on to prevent over requesting the API.